scripts/run_fcst.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+"""
+The run script for run_fcst.
+"""
+
+import datetime as dt
+import logging
+import os
+import sys
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+
+from uwtools.api.config import get_yaml_config
+from uwtools.api.fs import link as uwlink
+from uwtools.api.fv3 import FV3
+from uwtools.api.logging import use_uwtools_logger
+from uwtools.api.template import render
+from uwtools.api.upp import UPP
+
+
+def _walk_key_path(config, key_path):
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError:
+            logging.error(f"Bad config path: {pathstr}")
+            raise
+        if not isinstance(subconfig, dict):
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
```

Comment:
I hadn't thought of exposing the `uwtools` version. Hmm... We could consider something like `uwtools.api.support` or `uwtools.api.utils` where we could expose otherwise-internal functions that we determine are more widely useful. Of course, that increases the API surface area and the risk of breaking changes when we need to change previously-internal functionality to suite still-internal needs. I think maybe we should keep this in mind but maybe wait until we have some kind of critical mass -- like 3 or more functions `uwtools` functions that we find ourselves reimplementing in apps -- before going this route.

I was thinking of just something alongside the other scripts that they could `import`.